gprofiler/profilers/perf.py

```diff
@@ -257,15 +268,22 @@ def snapshot(self) -> ProcessToProfileData:
             perf.switch_output()
 
         return {
-            # TODO generate appids for non runtime-profiler processes here
-            k: ProfileData(v, None, self._get_metadata(k))
+            k: self._generate_profile_data(v, k)
             for k, v in merge.merge_global_perfs(
                 self._perf_fp.wait_and_script() if self._perf_fp is not None else None,
                 self._perf_dwarf.wait_and_script() if self._perf_dwarf is not None else None,
                 self._insert_dso_name,
             ).items()
         }
 
+    def _generate_profile_data(self, stacks: StackToSampleCount, pid: int) -> ProfileData:
+        metadata = self._get_metadata(pid)
+        if type(metadata) is dict and "node_version" in metadata.keys():
```

Comment:
```suggestion
        if metadata is not None and "node_version" in metadata:
```
1. The check `type(metadata) is dict` tells the reader "metadata might be a dict and might be other things" while in reality "metadata" is a dict or `None`.
2. No need for `.keys()` - `x in dict` does that.